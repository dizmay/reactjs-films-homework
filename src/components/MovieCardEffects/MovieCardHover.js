import React, { useState } from 'react';
import PropTypes from 'prop-types';
import MovieCardInfo from './MovieCardInfo';
import ButtonElement from '../ButtonElement/ButtonElement';
import styles from './MovieCardHover.scss';

const MovieCardHover = ({ hover, showTrailer, ...props }) => {
  const [info, setInfo] = useState(false);
  const showInfo = () => {
    setInfo(!info);
  };
  return (
    <>
      {
        info
          ? (
            <MovieCardInfo
              info={info}
              showInfo={showInfo}
              movieInfo={props}
              showTrailer={showTrailer}
            />
          )
          : (
            <div className={hover ? styles.hover__container : styles.hover_hideContainer}>
              <div className={styles.hover__watchNow}>
                <button type="button" onClick={showTrailer} className={styles.watchNow__btn}>&#9658;</button>
                <span className={styles.watchNow__title}>Watch Now</span>
              </div>
              <ButtonElement transparent clickFunc={showInfo}>View Info</ButtonElement>
            </div>
          )
      }
    </>
  );
};

MovieCardHover.defaultProps = {
  hover: false,
  showTrailer: null,
};

MovieCardHover.propTypes = {
  hover: PropTypes.bool,
  showTrailer: PropTypes.func,
};

export default MovieCardHover;
