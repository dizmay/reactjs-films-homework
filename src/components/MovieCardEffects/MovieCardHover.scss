* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.hover__container {
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  background: #000000b3;
}

.hover_hideContainer {
  display: none;
}

.hover__watchNow {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-bottom: 2rem;
  cursor: pointer;
}

.watchNow__btn {
  width: 6rem;
  height: 6rem;
  border-radius: 50%;
  border: none;
  outline: none;
  background: #00a3f5;
  color: #fff;
  font-size: 3rem;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-bottom: 1rem;
  cursor: pointer;
  padding-left: 0.5rem;
}

.watchNow__title {
  color: #fff;
  font-size: 1.5rem;
}
