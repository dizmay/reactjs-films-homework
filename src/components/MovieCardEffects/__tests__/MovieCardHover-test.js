import React from 'react';
import renderer from 'react-test-renderer';
import MovieCardHover from '../MovieCardHover';

it('renders correctly', () => {
  const tree = renderer.create(<MovieCardHover />).toJSON();
  expect(tree).toMatchSnapshot();
});
