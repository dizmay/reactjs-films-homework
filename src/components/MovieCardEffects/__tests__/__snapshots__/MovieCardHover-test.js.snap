// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="hover_hideContainer"
>
  <div
    className="hover__watchNow"
  >
    <button
      className="watchNow__btn"
      onClick={null}
      type="button"
    >
      â–º
    </button>
    <span
      className="watchNow__title"
    >
      Watch Now
    </span>
  </div>
  <button
    className="btn transparent"
    onClick={[Function]}
    type="button"
  >
    View Info
  </button>
</div>
`;
